src/components/Card/Card.js

```diff
@@ -1,13 +1,34 @@
 import React from "react";
+import { useNavigate } from "react-router-dom";
+import {
+  useAddToWishlistMutation,
+  useRemoveFromWishlistMutation,
+  useUserQuery
+} from "../../hooks";
 import "./Card.css";
 
 const Card = ({ product }) => {
-  const { name, price, discount, pictureLink, cartItem } = product;
+  const { mutate: addToWishlist } = useAddToWishlistMutation();
+  const { mutate: removeFromWishlist } = useRemoveFromWishlistMutation();
+  const { data } = useUserQuery();
+  const { user } = data ?? {};
```

Comment:
you can destructure this way: const {data:{user}} = useUserQuery()

Reply:
yes Rahul, but when `data` is `undefined` here, it throw an error, that is why i have destructured like this, thanks